cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -938,6 +938,91 @@ mod test {
         }};
     }
 
+    trait ErrorsHelper {
+        fn expect_errors_at_line_col(self, src: &str, es: &[(YaccGrammarErrorKind, usize, usize)]);
+        fn expect_error_at_line(self, src: &str, kind: YaccGrammarErrorKind, line: usize);
+        fn expect_error_at_line_col(
+            self,
+            src: &str,
+            kind: YaccGrammarErrorKind,
+            line: usize,
+            col: usize,
+        );
+        fn expect_error_at_line_col_extra(
+            self,
+            src: &str,
+            kind: YaccGrammarErrorKind,
+            line: usize,
+            col: usize,
+            extra_spans: &[Span],
+        );
+    }
+
+    impl ErrorsHelper for Result<GrammarAST, Vec<YaccGrammarError>> {
+        fn expect_error_at_line(self, src: &str, kind: YaccGrammarErrorKind, line: usize) {
+            match self.as_ref().map_err(Vec::as_slice) {
+                Ok(_) => panic!("Incorrect %prec parsed"),
+                Err([e]) if e.kind == kind && line_of_offset(src, e.span().start()) == line => {
+                    assert_eq!(e.extra_spans(), &[])
+                }
+                Err(e) => incorrect_errs!(src, e),
+            }
+        }
+
+        fn expect_error_at_line_col(
+            self,
+            src: &str,
+            kind: YaccGrammarErrorKind,
+            line: usize,
+            col: usize,
+        ) {
+            match self.as_ref().map_err(Vec::as_slice) {
+                Ok(_) => panic!("Incorrect token parsed"),
+                Err([e]) if e.kind == kind && line_col!(src, e.span()) == (line, col) => {
+                    assert_eq!(e.extra_spans(), &[])
+                }
+                Err(e) => incorrect_errs!(src, e),
+            }
+        }
+
+        fn expect_error_at_line_col_extra(
+            self,
+            src: &str,
+            kind: YaccGrammarErrorKind,
+            line: usize,
+            col: usize,
+            extra_spans: &[Span],
```

Comment:
Now that we removed the `extra_spans` function in favor of a single span, this function needs to be rethought.

I think `expect_error_kind_with_lines_cols(&self, &str, kind, &[(usize, usize)])`
and `expect_multiple_errors(&self, &str, &[(kind, &[(usize, usize)]])` or something to that effect

Reply:
I tried to make the first more consistent with the new names etc,
https://github.com/softdevteam/grmtools/pull/322/commits/743ca1c49e05f811aaeec5d61f0ee9b6aaeecc93

I removed the second fn until we have a use for it, expect to readd it in a subsequent patch when we return multiple errors.
https://github.com/softdevteam/grmtools/pull/322/commits/cc445e189b3d9568a85712272138b1bfbd0a3223